apollo-portal/src/main/java/com/ctrip/framework/apollo/portal/spi/springsecurity/SpringSecurityUserService.java

```diff
@@ -75,29 +75,52 @@ public void createOrUpdate(UserPO user) {
       managedUser.setPassword(newPassword);
       managedUser.setEmail(user.getEmail());
       managedUser.setUserDisplayName(user.getUserDisplayName());
+      managedUser.setEnabled(user.getEnabled());
       userRepository.save(managedUser);
     }
   }
 
+  @Transactional
+  public void changeEnabled(UserPO user) {
+    String username = user.getUsername();
+    UserPO managedUser = userRepository.findByUsername(username);
+    if (user.getEnabled() == 1) {
+      managedUser.setEnabled(0);
+    } else {
+      managedUser.setEnabled(1);
+    }
```

Comment:
This logic is a little weird. I think it's better to pass the desired enabled state and simply update it here. 
```suggestion
    managedUser.setEnabled(user.getEnabled());
```